File: katt/kingdom/attachments/python2/grader.py
import grader
import kingdom
import sys

_parts = []

def part(R):
  _parts.append(R)


def _grader_init():
  N, P = map(int, sys.stdin.readline().split())
  C = [ int(t) for t in sys.stdin.readline().strip().split() ]
  F = [ int(t) for t in sys.stdin.readline().strip().split() ]
  T = [ int(t) for t in sys.stdin.readline().strip().split() ]

  print(kingdom.division(N, P, C, F, T))
  print(len(_parts))
  for part in _parts:
    sys.stdout.write(str(len(part)))
    for i, e in enumerate(part):
      sys.stdout.write(" %d" % e)
    sys.stdout.write("\n")

if __name__ != "__main__":                                                          
  grader._grader_init()

File: katt/kingdom/attachments/java/grader.java
import java.io.*;
import java.util.*;

public class grader {


	public static void main(String... args) throws IOException {
		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
		PrintWriter w = new PrintWriter(new BufferedOutputStream(System.out));
		String[] np = r.readLine().split(" ");
		int N = Integer.parseInt(np[0]),
				P = Integer.parseInt(np[1]);

		int[] C = new int[N],
			F = new int[N - 1],
			T = new int[N - 1];

		String[]
			partsC = r.readLine().split(" "),
						 partsF = r.readLine().split(" "),
						 partsT = r.readLine().split(" ");

		for (int i = 0; i < N; ++i) {
			C[i] = Integer.parseInt(partsC[i]);
		}
		for (int i = 0; i < N - 1; ++i) {
			F[i] = Integer.parseInt(partsF[i]);
			T[i] = Integer.parseInt(partsT[i]);
		}
		w.println(kingdom.division(N, P, C, F, T));
		for (int p : lib.R) {
			w.print(p);
			w.print(" ");
		}
		w.close();
	}

	public static class lib {

		private static int[] R;
		public static void parts(int[] R) {
			lib.R = R;
		}

	}

}
